sql/core/src/main/scala/org/apache/spark/sql/execution/python/FlatMapGroupsInPandasExec.scala

```diff
@@ -145,7 +145,8 @@ case class FlatMapGroupsInPandasExec(
         sessionLocalTimeZone,
         pythonRunnerConf).compute(grouped, context.partitionId(), context)
 
-      columnarBatchIter.flatMap(_.rowIterator.asScala).map(UnsafeProjection.create(output, output))
```

Comment:
@HyukjinKwon I could be wrong, but I believe `UnsafeProjection.create(output, output)` creates the function before it is executed by map, so it's only done once. If it was `map(_ => UnsafeProjection.create(output, output)` then it would get created a bunch of times. Maybe I am missing something tho?

Reply:
hmm, I think so too.

Reply:
My bad. Agree to above comments. I'd still be in favor of explicitly moving it out to remove confusion.

Reply:
I think the patch is cleaner but perhaps the PR title is not accurate?

Reply:
D'oh! yes. It was my mistake.